modules/datomic/src/modular/datomic.clj

```diff
@@ -53,7 +53,7 @@
   component/Lifecycle
   (start [this]
     (with-open [rdr (java.io.PushbackReader. (io/reader res))]
-      @(d/transact (:connection this)
+      @(d/transact (get-in this [:connection :connection])
```

Comment:
There should be a function to access the connection from a `DatomicConnection` component. This function hides the actual implementation:

```
(defn conn [connection-comp]
 (get-in connection-comp connection))
```

And here you'd use `@(d/transact (conn (:connection this))`

I'm not sure if `DatomicConnection` is the best name, ConnectionHolder or like that might be better. But that's out of scope for this issue.